code/datums/brain_damage/brain_trauma.dm

```diff
@@ -61,4 +72,69 @@
 
 //Called when hugging. expand into generally interacting, where future coders could switch the intent?
 /datum/brain_trauma/proc/on_hug(mob/living/hugger, mob/living/hugged)
-	return
+	if(resilience >= TRAUMA_RESILIENCE_WOUND)
+		return
+	if(!COOLDOWN_FINISHED(src, hug_therapy_cd))
+		return
+	COOLDOWN_START(src, hug_therapy_cd, hug_therapy_cd_time)
+	var/cure_chance = random_cure_chance / 6
+	if(HAS_TRAIT(hugger.mind, TRAIT_PSYCH))
+		cure_chance *= 3.5
+	if(HAS_TRAIT(hugger, TRAIT_FRIENDLY))
+		cure_chance *= 1.25
+	if(iscarbon(hugged))
+		var/mob/living/carbon/C = hugged
+		if(C.hypnosis_vulnerable())
+			cure_chance *= 1.5
+	if(prob(cure_chance))
+		qdel(src) // Sometimes, all you need is a good hug..
```

Comment:
Code is duplicated starting from the ``iscarbon`` check, make a new proc where you pass in the ``cure_chance`` and ``hugger``